docs(migration): state how to register components in `content` folder as global components

BayBreezy opened the pull request:
In v2, this was done automatically. It has to be explicitly added in v3.

The issue raised can be found here #2930 

<!---
☝️ PR title should follow conventional commits (https://conventionalcommits.org)
-->

### 🔗 Linked issue

### ❓ Type of change

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

While migrating from v2, I noticed that all the components that were once global and could be passed as a string to the `<component />` element were no longer being displayed. The components created under `components/content` were no longer automatically registered as global components.

This PR just adds this note to the migration page so that users know that they have to explicitly mark components within the `components/content` folder as global. 

<!-- Describe your changes in detail -->
<!-- Why is this change required? What problem does it solve? -->
<!-- If it resolves an open issue, please link to the issue here. For example "Resolves #1337" -->

### 📝 Checklist

<!-- Put an `x` in all the boxes that apply. -->
<!-- If your change requires a documentation PR, please link it appropriately -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

ineshbose commented:
Aside from hook, the file names can be renamed to include `.global.vue` suffix, right?

farnabaz reviewed (changes requested):
Thanks for the PR. However, this action is not necessary for all users, in your case it is required because you are using these components as dynamic components.

Instead of adding a section, let's add a note in `Rendering Components` section and notify about changing from global to non-global. And link to [Nuxt Custom Components Directories](https://nuxt.com/docs/guide/directory-structure/components#custom-directories) for more details about Nuxt components system.

BayBreezy commented:
> Aside from hook, the file names can be renamed to include `.global.vue` suffix, right?

Yes. You can do that too. For me, there are too many global components to do this.

BayBreezy commented:
> Nuxt Custom Components Directories

I made the changes just now. 
Removed the section and just added another point under the Rendering Components section.

farnabaz commented on docs/content/docs/1.getting-started/4.migration.md:
```diff
@@ -30,6 +30,7 @@ Don't worry, you don't need to modify your content files. We made sure that Cont
 - We consolidated the `ProsePre`, `ProseCode`, and `ProseCodeInline` components
   - `ProseCodeInline` is now `ProseCode`
   - `ProseCode` is now `ProsePre`
+- Components created under the `components/content` directory are no longer automatically registered as global components. You need to register them manually in your Nuxt app. Check out the [Nuxt - Custom Components Directories](https://nuxt.com/docs/guide/directory-structure/components#custom-directories) documentation for more information on how to do so.
```
```suggestion
- Components created under the `components/content` directory are no longer automatically registered as global components. If you use [dynamic rendering](https://vuejs.org/guide/essentials/component-basics.html#dynamic-components) to render these components outside markdown files, you must manually register them in your Nuxt app. Check out the [Nuxt - Custom Components Directories](https://nuxt.com/docs/guide/directory-structure/components#custom-directories) documentation for more information on how to do so.
```

farnabaz reviewed (approved):
Thanks

Mathias265 commented:
Beautifull work
